From ba632fde62d50cb2641010b29e6f11bf62862922 Mon Sep 17 00:00:00 2001
From: Noriko Hosoi <nhosoi@redhat.com>
Date: Fri, 8 Nov 2013 13:58:14 -0800
Subject: [PATCH 46/49] Ticket #47589 - Winsync replica initialization and
 incremental updates from DS to AD fails on RHEL7

Bug description: Cherry-picking the fix for  "Ticket #47492 - PassSync
removes User must change password flag on the Windows side" (commit
8d34f77f6d8d3c83dce1f29e6df709df1adef09d) dropped one line to set
suffix in map_entry_dn_outbound.

Fix description: This patch recovers the suffix setting code.

Reviewed by rmeggins (Thank you, Rich!)

(cherry picked from commit 9df31ed14bf8c9e2224a21e0fc544fffec637a58)
---
 ldap/servers/plugins/replication/windows_protocol_util.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ldap/servers/plugins/replication/windows_protocol_util.c b/ldap/servers/plugins/replication/windows_protocol_util.c
index 69db5a0..5559f7e 100644
--- a/ldap/servers/plugins/replication/windows_protocol_util.c
+++ b/ldap/servers/plugins/replication/windows_protocol_util.c
@@ -3502,7 +3502,7 @@ map_entry_dn_outbound(Slapi_Entry *e,
 	char *guid = NULL;
 	Slapi_DN *new_dn = NULL;
 	int is_nt4 = windows_private_get_isnt4(prp->agmt);
-	const char *suffix = NULL;
+	const char *suffix = slapi_sdn_get_dn(windows_private_get_windows_subtree(prp->agmt));
 	Slapi_Entry *remote_entry = NULL;
 
 	if (NULL == e) {
-- 
1.8.1.4

